cpp/ql/src/Diagnostics/Files.ql

```diff
@@ -0,0 +1,10 @@
+/**
+ * @name Total source files
+ * @description The total number of source files.
+ * @kind metric
+ * @id cpp/metrics/files
```

Comment:
These queries are in a `Diagnostics` directory, and I think that describes them well. Therefore I'd write `diagnostics` or `diagnostic` rather than `metrics` in the query ids. This of course has to be coordinated across languages. What do you think, @adityasharad ?

Reply:
In our design for extractor diagnostics we used the kind `diagnostic`, so it's up to you whether you want to give the same directory name and ID prefix for metrics+diagnostics, or you want to distinguish between those two. The IDs will be visible in the SARIF.

I have a slight preference for distinguishing them but honestly do not feel strongly about this.

Reply:
@adityasharad Is the audience for extractor diagnostics different from the audience for lines-of-code metrics? Aren't they both for determining whether CodeQL is extracting the right stuff? I suppose that they're two different levels of diagnostic (look at the lines-of-code summary, and if that looks wrong, look at extractor diagnostics).

I don't like using "metrics", because to me that implies the sort of metrics that ODASA used to do (Halstead, Cyclomatic Complexity, etc.). How about using something like `@id cpp/summary/lines-of-code` for the queries implemented here?

Reply:
Sounds good!